compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/JAREntries.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.ballerinalang.compiler;
+
+import io.ballerina.projects.ProjectException;
+import org.apache.commons.compress.archivers.jar.JarArchiveEntry;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.jar.Attributes;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+import java.util.jar.Manifest;
+
+/**
+ * This class holds .class files as entries of a program jar file and convert them to a ByteArrayOutputStream.
+ *
+ * @since 2201.10.0
+ */
+public class JAREntries {
+    private ByteArrayOutputStream byteArrayOutputStream;
+    private JarOutputStream entries;
+
+    public JAREntries(String mainClassName) {
+        this.byteArrayOutputStream = new ByteArrayOutputStream();
+        Manifest manifest = new Manifest();
+        Attributes mainAttributes = manifest.getMainAttributes();
+        mainAttributes.put(Attributes.Name.MANIFEST_VERSION, "1.0");
+        if (mainClassName != null) {
```

Comment:
I guess we need to check for the empty string value as well.

Reply:
I don't think we need to handle it because this class will be used especially for this case and `getModuleLevelClassName(module.packageID, MODULE_INIT_CLASS_NAME, ".")` never returns empty or null. Will remove the `null` check as well